src/utils/daos_hdlr.c

```diff
@@ -989,7 +989,7 @@ fs_copy_file(struct cmd_args_s *ap,
 	     const char *dst_path)
 {
 	int src_flags		= O_RDONLY;
-	int dst_flags		= O_CREAT | O_WRONLY;
+	int dst_flags		= O_CREAT | O_TRUNC | O_WRONLY;
```

Comment:
This will have a performance impact.

Another option might be to use O_EXCL here, and retry on EEXISTS which would allow better reporting to the user, and would have a different performance impact - I'm not sure which would be faster for the non-contended case.

Reply:
How does retrying help? The file could still exist. The idea with the ticket was to match the performance of `cp` and `mpifileutils/dcp`. For example:
```
echo src > src
echo dst > dst
cp src dst
# no complaints - overrides
```

Reply:
I just checked the time to copy a directory with 20,000 empty files with and without "O_TUNC". The time difference is less than 4%.

Reply:
When the destination is located on "/dev/shm/", the time difference is less than 1%.

Reply:
> How does retrying help? The file could still exist.

The retry would have to replace O_EXCL with O_TRUNC which should then work.

Reply:
No problems here, just wanted to know the implications had been thought through.